themes/ddbasic/scripts/ddbasic.common.js

```diff
@@ -20,11 +20,21 @@
         text.push(Drupal.t('Opening hours'));
       }
 
-      $('<a />', {
-        'class' : 'opening-hours-toggle js-opening-hours-toggle js-collapsed collapsed',
-        'href' : Drupal.t('#toggle-opening-hours'),
-        'text' : text.join(', ')
-      }).insertBefore(this);
+      if (Drupal.settings.ding_ddbasic_opening_hours && Drupal.settings.ding_ddbasic_opening_hours.expand_all_libraries) {
```

Comment:
A better way to test if properties exits would be to use hasOwnProperty().

Reply:
You could also preform the test once (as its used more than once in the code) and put it into a variable.